storage/memory.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const LRUCache = require('mnemonist/lru-cache')
+const StorageInterface = require('./interface')
+
+const DEFAULT_CACHE_SIZE = 1024
+
+/**
+ * @typedef StorageMemoryOptions
+ * @property {?number} [size=1024]
+ * @property {Logger} log
+ */
+
+class StorageMemory extends StorageInterface {
+  /**
+   * in-memory storage
+   * @param {StorageMemoryOptions} options
+   */
+  constructor (options = {}) {
+    // TODO validate options
+    // logger is mandatory
+    super(options)
+    this.store = new LRUCache(options.size || DEFAULT_CACHE_SIZE)
+    this.references = new Map()
+    this.log = options.log
+  }
+
+  async get (key) {
+    const entry = this.store.get(key)
+    if (entry) {
+      if (entry.expires > Date.now()) {
+        return entry.value
+      }
+      this.store.set(key, undefined)
+    }
+  }
+
+  async remove (key) {
+    console.log('storage.remove', key)
+    this.store.set(key, undefined)
+  }
+
+  async set (key, value, ttl, references) {
+    this.store.set(key, { value, expires: Date.now() + ttl })
+
+    if (!references) {
+      return
+    }
+    for (let i = 0; i < references.length; i++) {
+      const reference = references[i]
+      let keys = this.references.get(reference)
```

Comment:
I think we should validate that all references are still in the cache.

Alterntively, we should get a callback with the evicted value.